[BUG] 9.6.x: Unresolved reference 'handlePushMessage' with Intercom Android SDK 18.0.0

## Description

`intercom_flutter` 9.6.0 bumped the Intercom Android SDK to `18.0.0`, but `IntercomFlutterPlugin.kt` still references `Intercom.client().handlePushMessage()` which was removed in SDK 18.0.0.

## Error

```
IntercomFlutterPlugin.kt:206:27 Unresolved reference 'handlePushMessage'.

FAILURE: Build failed with an exception.
Execution failed for task ':intercom_flutter:compileDebugKotlin'.
```

## Location

https://github.com/v3rm0n/intercom_flutter/blob/master/intercom_flutter/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt#L206

```kotlin
"handlePushMessage" -> {
    Intercom.client().handlePushMessage()  // This method no longer exists in SDK 18.0.0
    result.success("Push message handled")
}
```

## Reproduction

1. Use `intercom_flutter: ^9.5.0` (resolves to 9.6.1)
2. Build for Android
3. Compilation fails on `handlePushMessage`

## Environment

- Flutter 3.38.x
- `compileSdk 36`
- `intercom_flutter: 9.6.1`

## Workaround

Pin to `intercom_flutter: ">=9.5.0 <9.6.0"` to avoid the broken Intercom Android SDK 18.0.0.

--- a/intercom_flutter/CHANGELOG.md
+++ b/intercom_flutter/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## 9.6.2
+
+* Removed deprecated `handlePushMessage` API (removed from native Intercom SDK)
+
 ## 9.6.1
 
 * Bump Intercom iOS SDK version to [19.5.6](https://github.com/intercom/intercom-ios/releases/tag/19.5.6)

--- a/intercom_flutter/README.md
+++ b/intercom_flutter/README.md
@@ -144,7 +144,6 @@ But you can pre-define some Intercom settings, if you want (optional).
 - [ ] unreadConversationCount
 - [ ] setInAppMessagesVisibility
 - [ ] sendTokenToIntercom
-- [ ] handlePushMessage
 - [ ] isIntercomPush
 - [ ] handlePush
 - [ ] displayCarousel

--- a/intercom_flutter/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt
+++ b/intercom_flutter/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt
@@ -202,10 +202,6 @@ class IntercomFlutterPlugin : FlutterPlugin, MethodCallHandler, EventChannel.Str
           result.success("Token sent to Intercom")
         }
       }
-      "handlePushMessage" -> {
-        Intercom.client().handlePushMessage()
-        result.success("Push message handled")
-      }
       "displayMessageComposer" -> {
         if (call.hasArgument("message")) {
           Intercom.client().displayMessageComposer(call.argument("message"))

--- a/intercom_flutter/ios/intercom_flutter/Sources/intercom_flutter/IntercomFlutterPlugin.m
+++ b/intercom_flutter/ios/intercom_flutter/Sources/intercom_flutter/IntercomFlutterPlugin.m
@@ -170,9 +170,6 @@ id unread;
             result(@"Logged event");
         }
     }
-    else if([@"handlePushMessage" isEqualToString:call.method]) {
-        result(@"No op");
-    }
     else if([@"displayMessageComposer" isEqualToString:call.method]) {
         NSString *message = call.arguments[@"message"];
         [Intercom presentMessageComposer:message];

--- a/intercom_flutter/lib/intercom_flutter.dart
+++ b/intercom_flutter/lib/intercom_flutter.dart
@@ -210,16 +210,6 @@ class Intercom {
     return IntercomFlutterPlatform.instance.sendTokenToIntercom(token);
   }
 
-  /// When a user taps on a push notification Intercom hold onto data
-  /// such as the URI in your message or the conversation to open.
-  ///
-  /// When you want Intercom to act on that data, use this method.
-  @Deprecated(
-      "Calling this API is no longer required. Intercom will directly open the chat screen when a push notification is clicked.")
-  Future<void> handlePushMessage() {
-    return IntercomFlutterPlatform.instance.handlePushMessage();
-  }
-
   /// To open the Intercom messenger to the composer screen with [message]
   /// field pre-populated.
   Future<void> displayMessageComposer(String message) {

--- a/intercom_flutter/pubspec.yaml
+++ b/intercom_flutter/pubspec.yaml
@@ -1,7 +1,7 @@
 name: intercom_flutter
 description: Flutter plugin for Intercom integration. Provides in-app messaging
   and help-center Intercom services
-version: 9.6.1
+version: 9.6.2
 homepage: https://github.com/v3rm0n/intercom_flutter
 
 dependencies:

--- a/intercom_flutter_platform_interface/lib/intercom_flutter_platform_interface.dart
+++ b/intercom_flutter_platform_interface/lib/intercom_flutter_platform_interface.dart
@@ -189,16 +189,6 @@ abstract class IntercomFlutterPlatform extends PlatformInterface {
     throw UnimplementedError('sendTokenToIntercom() has not been implemented.');
   }
 
-  /// When a user taps on a push notification Intercom hold onto data
-  /// such as the URI in your message or the conversation to open.
-  ///
-  /// When you want Intercom to act on that data, use this method.
-  @Deprecated(
-      "Calling this API is no longer required. Intercom will directly open the chat screen when a push notification is clicked.")
-  Future<void> handlePushMessage() {
-    throw UnimplementedError('handlePushMessage() has not been implemented.');
-  }
-
   /// To open the Intercom messenger to the composer screen with [message]
   /// field pre-populated.
   Future<void> displayMessageComposer(String message) {

--- a/intercom_flutter_platform_interface/lib/method_channel_intercom_flutter.dart
+++ b/intercom_flutter_platform_interface/lib/method_channel_intercom_flutter.dart
@@ -175,13 +175,6 @@ class MethodChannelIntercomFlutter extends IntercomFlutterPlatform {
     await _channel.invokeMethod('sendTokenToIntercom', {'token': token});
   }
 
-  @Deprecated(
-      "Calling this API is no longer required. Intercom will directly open the chat screen when a push notification is clicked.")
-  @override
-  Future<void> handlePushMessage() async {
-    await _channel.invokeMethod('handlePushMessage');
-  }
-
   @override
   Future<void> displayMessageComposer(String message) async {
     await _channel.invokeMethod('displayMessageComposer', {'message': message});

--- a/intercom_flutter_web/README.md
+++ b/intercom_flutter_web/README.md
@@ -21,7 +21,6 @@ But you can pre-define some Intercom settings, if you want (optional), such as `
 - unreadConversationCount
 - setInAppMessagesVisibility
 - sendTokenToIntercom
-- handlePushMessage
 - isIntercomPush
 - handlePush
 - displayCarousel

